pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -59,7 +81,405 @@ func (r *Builder) mapDisks(vm *model.Workload, persistentVolumeClaims []core.Per
+
+	kNetworks = append(kNetworks, kNetwork)
+	kInterfaces = append(kInterfaces, kInterface)
+	object.Template.Spec.Networks = kNetworks
+	object.Template.Spec.Domain.Devices.Interfaces = kInterfaces
+
+	return nil
+}
+
+// Build tasks.
+func (r *Builder) Tasks(vmRef ref.Ref) (list []*plan.Task, err error) {
+	vm := &model.Workload{}
+	err = r.Source.Inventory.Find(vm, vmRef)
+	if err != nil {
+		err = liberr.Wrap(
+			err,
+			"VM lookup failed.",
+			"vm",
+			vmRef.String())
+	}
+
+	for _, va := range vm.AttachedVolumes {
+		gb := int64(va.Size)
+		list = append(
+			list,
+			&plan.Task{
+				Name: fmt.Sprintf("%s-%s", r.Migration.Name, va.ID),
+				Progress: libitr.Progress{
+					Total: gb * 1024,
+				},
+				Annotations: map[string]string{
+					"unit": "MB",
+				},
+			})
+	}
+
+	return
+}
+
+// Create DataVolume certificate configmap.
+func (r *Builder) ConfigMap(_ ref.Ref, in *core.Secret, object *core.ConfigMap) (err error) {
+	return
+}
+
+func (r *Builder) DataVolumes(vmRef ref.Ref, secret *core.Secret, configMap *core.ConfigMap, dvTemplate *cdi.DataVolume) (dvs []cdi.DataVolume, err error) {
+	return nil, nil
+}
+
+// Build tasks.
+func (r *Builder) TemplateLabels(vmRef ref.Ref) (labels map[string]string, err error) {
+	return nil, nil
+}
+
+// Return a stable identifier for a DataVolume.
+func (r *Builder) ResolveDataVolumeIdentifier(dv *cdi.DataVolume) string {
+	return ""
+}
+
+// Return a stable identifier for a PersistentDataVolume
+func (r *Builder) ResolvePersistentVolumeClaimIdentifier(pvc *core.PersistentVolumeClaim) string {
+	return ""
+}
+
+// Build the DataVolume credential secret.
+func (r *Builder) Secret(_ ref.Ref, in, object *core.Secret) (err error) {
+	// TODO: complete
+	object.StringData = map[string]string{
+		"accessKeyId": string(in.Data["user"]),
+		"secretKey":   string(in.Data["password"]),
+	}
+	return
+}
+
+func (r *Builder) PodEnvironment(_ ref.Ref, _ *core.Secret) (env []core.EnvVar, err error) {
+	return
+}
+
+func (r *Builder) PersistentVolumeClaimWithSourceRef(da interface{}, storageName *string, populatorName string, accessModes []core.PersistentVolumeAccessMode, volumeMode *core.PersistentVolumeMode) *core.PersistentVolumeClaim {
+	image := da.(*openstack.Image)
+	apiGroup := "forklift.konveyor.io"
+	return &core.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      image.ID,
+			Namespace: r.Plan.Spec.TargetNamespace,
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			AccessModes: accessModes,
+			Resources: core.ResourceRequirements{
+				Requests: map[core.ResourceName]resource.Quantity{
+					core.ResourceStorage: *resource.NewQuantity(int64(float64(image.VirtualSize)*1.1), resource.BinarySI)},
+			},
+			StorageClassName: storageName,
+			VolumeMode:       volumeMode,
+			DataSourceRef: &core.TypedLocalObjectReference{
+				APIGroup: &apiGroup,
+				Kind:     "OpenstackVolumePopulator",
+				Name:     populatorName,
+			},
+		},
+		Status: core.PersistentVolumeClaimStatus{},
```

Comment:
and also same as on the other PR, this shouldn't be necessarily (the Status section)